src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/FunctionPointerOps.cs

```diff
@@ -63,44 +28,16 @@ public static unsafe IntPtr GetGenericMethodFunctionPointer(IntPtr canonFunction
 
             lock (s_genericFunctionPointerDictionary)
             {
-                var key = new GenericMethodDescriptorInfo
-                {
-                    MethodFunctionPointer = canonFunctionPointer,
-                    InstantiationArgument = instantiationArgument
-                };
-
-                uint index = 0;
-                if (!s_genericFunctionPointerDictionary.TryGetValue(key, out index))
+                ref IntPtr descriptor = ref CollectionsMarshal.GetValueRefOrAddDefault(s_genericFunctionPointerDictionary, (canonFunctionPointer, instantiationArgument), out bool exists);
+                if (!exists)
```

Comment:
```suggestion
                // Check for null descriptor instead of `exists` to handle the situation where `NativeMemory.Alloc` below threw
                // an exception during earlier attempt to this descriptor
                if (descriptor == IntPtr.Zero)
```